### PR TITLE
[mellanox]: Align platform API: change CPLD version representation

### PR DESCRIPTION
**Pull request by nazariig**
This PR introduces new CPLD version representation for Mellanox platforms:
CPLD<part_number>_REV<major_version><minor_version>

part_number - six digits
major_version - two digits
minor_version - two digits

Example: `CPLD000000_REV0500`

Note: if Part Number or Minor Version is missing, zeros will be printed

**- What I did**
* Updated CPLD version representation

**- How I did it**
* N/A

**- How to verify it**
1. make configure PLATFORM=mellanox
2. make target/sonic-mellanox.bin

**- Description for the changelog**

* N/A

**- Previous fwutil output**
```bash
root@sonic:/home/admin# fwutil show status
Chassis                 Module    Component    Version                  Description
----------------------  --------  -----------  -----------------------  ---------------------------------------
x86_64-mlnx_msn3800-r0  N/A       BIOS         0ACLH004_02.02.003_9600  BIOS - Basic Input/Output System
                                  CPLD         5.3.3.1                  CPLD - includes all CPLDs in the switch
```

**- New fwutil output**
```bash
root@sonic:/home/admin# fwutil show status
Chassis                 Module    Component    Version                   Description
----------------------  --------  -----------  ------------------------  ---------------------------------------
x86_64-mlnx_msn3800-r0  N/A       BIOS         0ACLH004_02.02.003_9600   BIOS - Basic Input/Output System
                                  CPLD1        CPLD000000_REV0500        CPLD - Complex Programmable Logic Device
                                  CPLD2        CPLD000000_REV0300        CPLD - Complex Programmable Logic Device
                                  CPLD3        CPLD000000_REV0300        CPLD - Complex Programmable Logic Device
                                  CPLD4        CPLD000000_REV0100        CPLD - Complex Programmable Logic Device
```

**- A picture of a cute animal (not mandatory but encouraged)**
```bash
      .---.        .-----------
     /     \  __  /    ------
    / /     \(  )/    -----
   //////   ' \/ `   ---
  //// / // :    : ---
 // /   /  /`    '--
//          //..\\
       ====UU====UU====
           '//||\\`
             ''``
```

### CODE REVIEW

**Comment by nazariig** [comment]
https://github.com/Azure/sonic-utilities/pull/831

**Comment by jleveque** [comment]
Retest vsimage please

**Comment by jleveque** [comment]
LGTM. @sujinmkang to review, also.

File: platform/mellanox/mlnx-platform-api/sonic_platform/component.py [hunk 1]
```diff
@@ -246,19 +261,26 @@ def get_firmware_version(self):
...
+        part_number_file = self.CPLD_PART_NUMBER_FILE.format(self.idx)
+        version_file = self.CPLD_VERSION_FILE.format(self.idx)
+        version_minor_file = self.CPLD_VERSION_MINOR_FILE.format(self.idx)
+
+        part_number = self._read_generic_file(part_number_file, self.CPLD_PART_NUMBER_MAX_LENGTH, True)
+        version = self._read_generic_file(version_file, self.CPLD_VERSION_MAX_LENGTH)
+        version_minor = self._read_generic_file(version_minor_file, self.CPLD_VERSION_MINOR_MAX_LENGTH, True)
+
+        if part_number is None:
+            part_number = ZERO
```
Comment (stephenxs) [inline platform/mellanox/mlnx-platform-api/sonic_platform/component.py]: Is it better to use a more meaningful name here? Like `DEFAULT_PART_NUMBER` and `DEFAULT_VERSION_MINOR`?

File: platform/mellanox/mlnx-platform-api/sonic_platform/component.py [hunk 2]
```diff
@@ -18,13 +18,15 @@
 except ImportError as e:
     raise ImportError(str(e) + "- required module not found")
 
-#components definitions
-COMPONENT_BIOS = "BIOS"
-COMPONENT_CPLD = "CPLD"
+EMPTY = ''
+ZERO = '0'
+COMMA = ','
```
Comment (stephenxs) [inline platform/mellanox/mlnx-platform-api/sonic_platform/component.py]: We don't need `COMMA`. 
It is not referenced. Even it is, to use the literal constant `,` is more straightforward IMO.

File: platform/mellanox/mlnx-platform-api/sonic_platform/component.py (hunk 1 above)
Comment (nazariig) [reply platform/mellanox/mlnx-platform-api/sonic_platform/component.py]: @stephenxs fixed

File: platform/mellanox/mlnx-platform-api/sonic_platform/component.py (hunk 2 above)
Comment (nazariig) [reply platform/mellanox/mlnx-platform-api/sonic_platform/component.py]: @stephenxs fixed

**Comment by nazariig** [comment]
retest default please

**Comment by nazariig** [comment]
retest vsimage please

**Comment by jleveque** [comment]
Retest vsimage please

**Pull request merged**